src/secrets/CommandEdit.ts

```diff
@@ -54,40 +55,85 @@ class CommandEdit extends CommandPolykey {
           },
           logger: this.logger.getChild(PolykeyClient.name),
         });
-        const response = await binUtils.retryAuthentication(
-          (auth) =>
-            pkClient.rpcClient.methods.vaultsSecretsGet({
-              metadata: auth,
-              nameOrId: secretPath[0],
-              secretName: secretPath[1],
-            }),
-          meta,
-        );
-        const secretContent = response.secretContent;
+
         const tmpDir = await fs.promises.mkdtemp(
           path.join(os.tmpdir(), 'polykey-'),
         );
         const tmpFile = path.join(tmpDir, 'pksecret');
-        await this.fs.promises.writeFile(tmpFile, secretContent);
+        const secretExists = await binUtils.retryAuthentication(
+          async (auth) => {
+            let exists: boolean = true;
+            try {
+              const response =
+                await pkClient.rpcClient.methods.vaultsSecretsGet({
+                  metadata: auth,
+                  nameOrId: secretPath[0],
+                  secretName: secretPath[1],
+                });
+              await this.fs.promises.writeFile(tmpFile, response.secretContent);
+            } catch (e) {
+              const [cause, _] = binUtils.remoteErrorCause(e);
+              if (cause instanceof vaultsErrors.ErrorSecretsSecretUndefined) {
+                exists = false;
+              } else {
+                throw e;
+              }
+            }
+            return exists;
+          },
+          meta,
+        );
+        // If the editor exited with a code other than zero, then execSync
+        // will throw an error. So, in the case of saving the file but the
+        // editor crashing, the program won't save the updated secret.
         execSync(`$EDITOR \"${tmpFile}\"`, { stdio: 'inherit' });
```

Comment:
Apparently when you invoke environment variables using `SVARIABLE` in `execSync`, it doesn't respect the variable value in `process.env`. For me, while I had set `process.env.EDITOR` to a script I made for testing the `secrets edit` command, the `execSync` kept opening my actual system editor.

However, logging the value of `process.env.EDITOR` showed me that the value was set correctly. So, when I replaced `$EDITOR` with `process.env.EDITOR`, the command started respecting the actual environment variable and using the script correctly.

It is weird, because `ExecSync` by default uses `NodeJS.ProcessEnv` for its environment variables, but for some reason, it didn't work with the tests.